src/Sales.cpp

```diff
@@ -0,0 +1,103 @@
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <string>
+#include <vector>
+#include "Sales.hpp"
+
+/*
+ * Constructor for base Sales class, initializes data to given values.
+ */
+Sale::Sale(const unsigned int id, const unsigned int sn, const std::string& d, const unsigned int as, const double sp, const double t, const std::string& b, const std::string& s)
+: identification(id), sale_number(sn), date(d), amount_sold(as), sale_price(sp), tax(t), buyer(b), seller(s)
+{
+    total_price = sale_price + (tax * sale_price);
+}
+
+/*
+ * Initilizes what file will be used to load/save the sales data to.
+ */
+SaleList::SaleList(const std::string& f)
+{
+    file_name = f;
+    offset = 0;
+}
+
+/*
+ * Creates a new sale to put in the sales vector.
+ */
+bool SaleList::add_sale(const unsigned int id, const unsigned int sn, const std::string& d, const unsigned int as, const double sp, const double t, const std::string& b, const std::string& s)
+{
+    if(id == 0 || sn == 0 || as == 0 || sp == 0 || d == "" || b == "" || s == "") return false;
```

Comment:
I think this should print something out to stderr here, or throw an exception if we think this should be an exit-level error.

Since this function would be presumably called by another function that's trying to read in the input, returning false is fine, but throwing an exception would be fine too.  I'd favor what you have now and just add a few things:
```
/* Check if input is bad; exit if so */
if(id == 0 || sn == 0 || as == 0 || sp == 0 || d == "" || b == "" || s == "")
{
    std::cerr << "Failed to read in sale with ID: " << id << ", continuing to read." << std::endl;
    return false;
}
```